rmf_fleet_adapter/include/rmf_fleet_adapter/agv/RobotUpdateHandle.hpp

```diff
@@ -128,11 +128,7 @@ class RobotUpdateHandle
     // TODO: Consider giving access to the participant schedule and
     // traffic negotiation
 
-    ~ActionExecution()
-    {
-      // Automatically trigger finished when this object dies
-      finished();
-    }
+    ~ActionExecution();
```

Comment:
Just one last minor thing: Let's add a line of documentation to this destructor explaining that `finished()` will automatically be triggered if it wasn't triggered already.